executor/builder.go

```diff
@@ -4932,3 +4962,45 @@ func (b *executorBuilder) buildCompactTable(v *plannercore.CompactTable) Executo
 		tikvStore:    tikvStore,
 	}
 }
+
+func setSnapshotOptions(b *executorBuilder, e Executor, snapshot kv.Snapshot) error {
+	sessVars := b.ctx.GetSessionVars()
+
+	switch v := e.(type) {
+	case *BatchPointGetExec:
```

Comment:
Seems for all executors this part of codes are the same. Use `e.base().runtimeStats` and remove `switch` ?
`

Reply:
v.stats cannot be got from e.base()